Bottom: f237b5daed5d0741bdd34d3ff61c5627a70c3a4f
Top:    459386463746d0a5de62f63298845b79d55e0928
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-13 02:16:34 +0100

hid/common/hidgl.c: Fix vertex array state preservation

Fixes a bug with crosshair attached objects not being visible
when the grid was rendered.

Reported-by: Colin D Bennett <colin@gibibit.com>
Closes-bug: lp-780958


---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index fb6b118..21322f7 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -66,8 +66,6 @@ float global_depth = 0;
 void
 hidgl_init_triangle_array (triangle_buffer *buffer)
 {
-  glEnableClientState (GL_VERTEX_ARRAY);
-  glVertexPointer (3, GL_FLOAT, 0, buffer->triangle_array);
   buffer->triangle_count = 0;
   buffer->coord_comp_count = 0;
 }
@@ -78,7 +76,11 @@ hidgl_flush_triangles (triangle_buffer *buffer)
   if (buffer->triangle_count == 0)
     return;
 
+  glEnableClientState (GL_VERTEX_ARRAY);
+  glVertexPointer (3, GL_FLOAT, 0, buffer->triangle_array);
   glDrawArrays (GL_TRIANGLES, 0, buffer->triangle_count * 3);
+  glDisableClientState (GL_VERTEX_ARRAY);
+
   buffer->triangle_count = 0;
   buffer->coord_comp_count = 0;
 }
@@ -140,6 +142,7 @@ hidgl_draw_grid (BoxType *drawn_area)
       points = realloc (points, npoints * 3 * sizeof (GLfloat));
     }
 
+  glPushAttrib ();
   glEnableClientState (GL_VERTEX_ARRAY);
   glVertexPointer (3, GL_FLOAT, 0, points);
